internal/metrics/metrics.go

```diff
@@ -248,6 +250,28 @@ func (t *TSDB) MaxTime() time.Time {
 	return fromDbTime(t.db.Head().MaxTime())
 }
 
+// HeadMinTime
+func (t *TSDB) HeadMinTime() time.Time {
+	return fromDbTime(t.db.Head().MinTime())
+}
+
+// BlockMinMax
+func (t *TSDB) BlocksMinMax() []time.Time {
+	blocks := t.db.Blocks()
+	blocksLen := len(blocks)
+	if blocksLen == 0 {
+		return []time.Time{}
+	}
+
+	result := make([]time.Time, blocksLen*2)
+	for i, block := range blocks {
+		result[i*2] = fromDbTime(block.MinTime())
+		result[i*2+1] = fromDbTime(block.MaxTime())
```

Comment:
Can a `BlocksMin` and a `BlocksMax` functions (i.e. split this function in two) make the code simpler to understand? After all, it's way way it's used in `skipEmptyRanges` [function](https://github.com/project-flotta/flotta-device-worker/blob/cb9d59bf4996acdf800f1b927b20713bcde5f0e6/internal/metrics/remote_write.go#L481). The [test](https://github.com/project-flotta/flotta-device-worker/blob/cb9d59bf4996acdf800f1b927b20713bcde5f0e6/internal/metrics/remote_write_test.go#L475) can also benefit from this change.

Reply:
I did not understand. I call 'block.MinTime'. So how is BlockMin different? can u suggest code edit? and.....does it really matter? cause this change is complex anyhow and requires a lot of attention.

Reply:
I'm sorry If my comment wasn't clear. 

You're using a slice like that contains `[block1_min, block1_max, block2_min, block2_max, ... ]` thus you need to iterate over the slice with`results[i*2]` and `results[i*2+1]`.
It isn't straightforward to understand the sense of the code, in my opinion.

I'm suggesting to use two different functions to retrieve two different slices (one for the max values and one for the min values) like these:

`func (t *TSDB) BlocksMin() []time.Time`
`func (t *TSDB) BlocksMax() []time.Time`

In this way, you can iterate over the slices in a "standard" way, the function goals will be clear and, at the same, time we will make life easier for a new contributor  :)

Reply:
In your example, you are referring to reading the min/max. But in my code I do not read from the slice. I create the slice. so, these functions will not help.
To me, this is very simple arithmetic that for us should be a no-brainer. I am actually, quite surprised that it is making so much noise. This is a very (very) common pattern in many libraries and functions. I can easily change my code based on everyone's comments. I just don't think it's the right way to perform our PR reviews. Think of the number of functions we add. I think it makes the code more difficult to track. Your call.

Reply:
I just gave you my opinion, feel free to make your own decisions

Reply:
@gciavarrini fixed